wurst/systems/core/ClassSelector.wurst

```diff
@@ -199,22 +202,19 @@ public class ClassSelector
         if gameConfig.isStartWithSpiritWard()
             troll.addItemById(ITEM_SPIRIT_WARD_KIT)
 
-    function handleRepick()
-        let sold = GetSoldUnit()
-        let owner = sold.getOwner()
-        let troll = owner.getTroll()
-        let trollId = troll.getTypeId()
-
-        setClassAvailable(trollId)
-        totem.addUnitToStock(trollId, 0, 1)
-        owner.removeTroll()
+    function handleTrollRemoval(player p)
+        let troll = p.getTroll()
+        if troll != null
+            let trollId = troll.getTypeId()
 
-        troll.remove()
-        sold.remove()
+            setClassAvailable(trollId)
+            totem.addUnitToStock(trollId, 1, 1)
+            p.removeTroll()
+            troll.remove()
 
-        owner
-        ..setGold(0)
-        ..setLumber(3)
+    function handleRandom(player p)
+        let trollId = getRandomTrollId()
+        IssueNeutralImmediateOrderById(p, totem, trollId)
```

Comment:
Does this work when the totem is a not a neutral building? I don't think it is when `-tb` is on but I could be wrong.

Reply:
ya works in my tests

Reply:
(with -tb not sure with allied shops, I'm not even sure u can make a owner player building cause shops are considered "neutral passive")

Reply:
Not all shops are neutral. Each race in melee has a shop they can build for example. These share most of the same functionality but only share control with allies instead of all allies.